TI-reports/2024/2024-Q4-BEST-WG.md

```diff
@@ -0,0 +1,248 @@
+
+- Our total 2024 enrollments for our "Developing Secure Software" course (LFD121) is 9,361. That was a dramatic increase from 2023 enrollments in LFD121, which was 6,658, and far exceeded our goal of 7,990.
+- Added labs for *all* of the "top tier" sections. These labs require no software installation; users just use their browsers. They also provide hints for those who are stuck.
+- Made various improvements, e.g., explained what open source software is, added various "story time" real-world examples from CrowdStrike and Meta, 
+
+#### Up Next
+
+- Continue to maintain content.
+
+
+### EDU.SIG
+
+#### Purpose
+
+- Deliver Baseline Secure Software Development Education and Certification to All.  Provide access to open and widely available education materials to all learners.
+Materials will be maximally accessible and easy to consume for all learners.
+
+#### Current Status
+
+- As noted above, LFD121 enrollment in 2024 was 9,361 and we completed all top-tier labs for it.
+- Draft course for managers developed, reviewed by LF Education, and their comments have been incorporated.
+
+#### Up Next
+
+- Complete the manager's course and have it posted.
+
+
+### Concise Guides
+
+#### Purpose
+
+- Artifacts that consolidate BEST practices in OSS software development and management techniques
+
+#### Current Status
+
+- Guides under active development:
+  - [C/C++ Compiler Option Hardening](#cc-compiler-option-hardening-guide)
+  - [Python Secure Coding Guide](#python-secure-coding-guide)
+
+#### Up Next
+
+- See status updates of respective guides
+
+
+### C/C++ Compiler Option Hardening Guide
+
+#### Purpose
+
+- Help C and C++ developers and those who compile C/C++ code, e.g., package maintainers, ensure that produced application binaries (libraries and executables) are equipped with security mechanisms provided by compilers against potential attacks and/or misbehavior.
+
+#### Current Status
+
+- Continued revision, updates, & enhancement, e.g., keeping the compiler options hardening guide up-to-date with upstream options additions and changes in GCC and Clang/LLVM and addressing feedback from Linux distribution communities.
+
+#### Up next
+
+- Microsoft MSVC guidance planned for 2025 (tracked in [BEST Issue 150](https://github.com/ossf/wg-best-practices-os-developers/issues/150))
+- Plan outreach activities for 2025, possibly talks aimed at C++ conferences.  
+
+
+### Python Secure Coding Guide
+
+#### Purpose
+
+- Help Python developers to create more secure code by explaining vulnerable and non-vulnerable coding patterns based on the CWE framework and rules.
+- Besides a description of each coding pattern, the guide includes executable code examples for each rule, which allow for an in-depth understanding of each pattern.
+
+#### Current Status
+
+- The group is working on adding more content for a broad range of CWE rules. The status is being tracked in issue [531](https://github.com/ossf/wg-best-practices-os-developers/issues/531).
```

Comment:
I found https://best.openssf.org/Secure-Coding-Guide-for-Python/, but it isn't linked from https://best.openssf.org/ or https://github.com/ossf/wg-best-practices-os-developers. Should I point people at this content? Or is it considered pre-release in its current form?

Reply:
Yes, @steiza, it is considered pre-release and therefore not yet linked on best.openssf.org. All content in the doc directory of the repo gets automatically rendered to https://best.openssf.org/Secure-Coding-Guide-for-Python/, so it shows up there, but is not linked on purpose.